BetterShardsBukkit/src/main/java/vg/civcraft/mc/bettershards/BetterShardsPlugin.java

```diff
@@ -150,20 +134,23 @@ private void addPlayerToTransit(final UUID uuid, final String server){
 
 			@Override
 			public void run() {
-				transit.remove(uuid);
-				Player player = Bukkit.getPlayer(uuid);
-				if (player != null && !player.isOnline()) {
+				if (isPlayerInTransit(uuid)) {
+					info(uuid + " failed to transit, was removed by timeout");
 					PlayerFailedToTransitEvent event = new PlayerFailedToTransitEvent(uuid, server);
 					Bukkit.getPluginManager().callEvent(event);
 				}
-				else {
-					PlayerEnsuredToTransitEvent event = new PlayerEnsuredToTransitEvent(uuid, server);
```

Comment:
It has to be called based on the callback of the other server and right now it doesnt keep track of which server the player was sent to after he's gone, so I cant call the event. I could turn the list into a map though which uses the server the player was sent to as value.